avocado/utils/lv_utils.py

```diff
@@ -0,0 +1,517 @@
+             lv_snapshot_name) in result.stdout):
+            raise Exception("The Logical volume %s is still active" %
+                            lv_name)
+        result = result.stdout.rstrip()
+
+    except Exception, ex:
+        # detect if merge of snapshot was postponed
+        # and attempt to reactivate the volume.
+        active_lv_pattern = re.escape("%s [active]" % lv_snapshot_name)
+        lvdisplay_output = process.run("lvdisplay").stdout
+        if ('Snapshot could not be found' in ex and
+                re.search(active_lv_pattern, lvdisplay_output) or
+                "The Logical volume %s is still active" % lv_name in ex):
+            LOGGER.debug(("Logical volume %s is still active! " +
+                          "Attempting to deactivate..."), lv_name)
+            lv_reactivate(vg_name, lv_name)
+            result = "Continuing after reactivation"
+        elif 'Snapshot could not be found' in ex:
+            LOGGER.error(ex)
+            result = "Could not revert to snapshot"
+        else:
+            raise ex
+    LOGGER.debug(result)
+
+
+def lv_revert_with_snapshot(vg_name, lv_name,
+                            lv_snapshot_name, lv_snapshot_size):
+    """
+    Perform Logical volume merge with snapshot and take a new snapshot.
+    """
+
+    lv_revert(vg_name, lv_name, lv_snapshot_name)
+    lv_take_snapshot(vg_name, lv_name, lv_snapshot_name, lv_snapshot_size)
+
+
+def lv_reactivate(vg_name, lv_name, timeout=10):
+    """
+    In case of unclean shutdowns some of the lvs is still active and merging
+    is postponed. Use this function to attempt to deactivate and reactivate
+    all of them to cause the merge to happen.
+    """
+    try:
+        process.run("lvchange -an /dev/%s/%s" % (vg_name, lv_name), shell=True)
+        time.sleep(timeout)
+        process.run("lvchange -ay /dev/%s/%s" % (vg_name, lv_name), shell=True)
+        time.sleep(timeout)
+    except process.CmdError:
+        LOGGER.error(("Failed to reactivate %s - please, " +
+                      "nuke the process that uses it first."), lv_name)
+        raise Exception("The Logical volume %s is still active" % lv_name)
+
+
+def lv_mount(vg_name, lv_name, mount_loc, create_filesystem=""):
+    """
+    Mount a Logical volume to a mount location.
+
+    The create_filesystem can be one of ext2, ext3, ext4, vfat or empty
+    if the filesystem was already created and the mkfs process is skipped.
+    """
+
+    try:
+        if create_filesystem:
+            result = process.run("mkfs.%s /dev/%s/%s" %
+                                 (create_filesystem, vg_name, lv_name),
+                                 shell=True)
+            LOGGER.debug(result.stdout.rstrip())
+        result = process.run("mount /dev/%s/%s %s" %
+                             (vg_name, lv_name, mount_loc), shell=True)
+    except process.CmdError, ex:
+        LOGGER.error(ex)
+        return False
```

Comment:
It'd be nicer to raise the exception and leave it on the caller to handle it. That way one get's finer granularity than True/False in code available. Anyway don't take this as requirement, only a suggestion for consideration.